core/src/mender-client.c

```diff
@@ -928,8 +928,10 @@ mender_client_update_work_function(void) {
     }
 
     {
-        char *artifact_type;
-        if (MENDER_OK == (ret = mender_storage_get_update_state(&update_state, &artifact_type))) {
+        char                 *artifact_type;
+        mender_update_state_t update_state_resume;
+        if (MENDER_OK == (ret = mender_storage_get_update_state(&update_state_resume, &artifact_type))) {
+            update_state = update_state_resume;
```

Comment:
I wrote this when I suspected the storage but finally I figured that the problem was that we were loading the MENDER_UPDATE_STATE_END state

> but better safe than sorry 👍

Yes! Several fixes in this PR are defensive programming style. As we might replace storage with something else in the future or it could even custom written is better to not assume too much.